main/src/main/java/cgeo/geocaching/network/HttpRequest.java

```diff
@@ -102,6 +104,10 @@ public Single<HttpResponse> request() {
     public <T> Single<T> requestJson(final Class<T> clazz) {
         headers("Accept", "application/json, text/javascript, */*; q=0.01");
         return requestInternal(r -> {
+            if (r.getStatusCode() == 429) {
```

Comment:
I suggest to move this code part to line 150 (into method `requestInternal`) instead of here. Then it will be used in all cases